Breaking test for trace search in ES.

garyd203 opened the pull request:
The Elastic Search implementation should be able to search traces
by serviceName of the server, when the server is not also a client
of another service.

For @adriancole

garyd203 committed:
Breaking test for trace search in ES.

The Elastic Search implementation should be able to search traces
by serviceName of the server, when the server is not also a client
of another service.

For @adriancole

codefromthecrypt commented:
Thanks Gary. Of course this change is welcome for anyone I dont mind
hogging it ;)

codefromthecrypt commented:
ok this actually breaks for everything except mysql and mem

codefromthecrypt commented:
in cassandra, this seems to be related to the server not setting timestamp (which it shouldn't)

codefromthecrypt commented:
nice find.. so far seems legit. I'll probably make sure this works for all or is understood why not.

Commit:
Fixes bug where server-side service names aren't indexed

While defending against accidentally overwriting the client's timestamp,
we accidentally clobbered the ability to refine by the server's service
name.

This fixes the issue and backports a test. It also buffs out some dodgy
code where we accidentally searched before 1970.

codefromthecrypt commented:
@garyd203 @sirtyro I added a commit to fix it

codefromthecrypt commented on zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/CassandraSpanConsumer.java:
```diff
@@ -103,43 +102,29 @@
-            futures.add(storeTraceServiceSpanName(serviceName, "", timestamp, span.duration, traceId));
-          }
-          futures.add(storeServiceSpanName(serviceName, span.name));
+        if (timestamp == null) continue; // QueryRequest.min/maxDuration needs timestamp
```
cc @llinder fyi this was just for sanity as indent level was blinding. no logic change

codefromthecrypt commented on zipkin/src/test/java/zipkin/storage/SpanStoreTest.java:
```diff
@@ -933,6 +935,34 @@ public void rawTrace_doesntPerformQueryTimeAdjustment() {
+   * when they didn't start a span.
+   */
+  @Test
+  public void traceIsSearchableBySRServiceName() throws Exception {
```
here's the new test
